File: src/editor/schema/branch.ts
import { Schema } from 'effect'

export const Branch = Schema.Struct({
  id: Schema.String,
  stream_id: Schema.String,
  name: Schema.NullOr(Schema.String),
  parent_branch_id: Schema.NullOr(Schema.String),
  fork_version: Schema.NullOr(Schema.Number),
  created_at: Schema.Number,
})
export type Branch = typeof Branch.Type

File: src-tauri/src/models/branch.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Branch {
    pub id: String,
    pub stream_id: String,
    pub name: Option<String>,
    pub parent_branch_id: Option<String>,
    pub fork_version: Option<i64>,
    pub created_at: i64,
}
